# Relatório 2
- Dataset: [aluguel.csv](./datasets/aluguel.csv)

Suponha que desejamos fazer uma análise exploratória apenas de _Imóveis Residenciais_ .

In [2]:
dataset = pd.read_csv('datasets/aluguel.csv', sep=';')
dataset

Tipo                Bairro  Quartos  Vagas  Suites  \
0                    Quitinete            Copacabana        1      0       0   
1                         Casa       Jardim Botânico        2      0       1   
2      Conjunto Comercial/Sala       Barra da Tijuca        0      4       0   
3                  Apartamento                Centro        1      0       0   
4                  Apartamento          Higienópolis        1      0       0   
...                        ...                   ...      ...    ...     ...   
32955                Quitinete                Centro        0      0       0   
32956              Apartamento           Jacarepaguá        3      1       2   
32957              Apartamento  São Francisco Xavier        2      1       0   
32958              Apartamento                Leblon        2      0       0   
32959  Conjunto Comercial/Sala                Centro        0      0       0   

       Area   Valor  Condominio    IPTU  
0        40  1700.0       500.0    60.0  
1       100  7000.0         NaN     NaN  
2       150  5200.0      4020.0  1111.0  
3        15   800.0       390.0    20.0  
4        48   800.0       230.0     NaN  
...     ...     ...         ...     ...  
32955    27   800.0       350.0    25.0  
32956    78  1800.0       800.0    40.0  
32957    48  1400.0       509.0    37.0  
32958    70  3000.0       760.0     NaN  
32959   250  6500.0      4206.0  1109.0  

[32960 rows x 9 columns]

## Tipos de Imóveis

In [5]:
dataset['Tipo']

0                      Quitinete
1                           Casa
2        Conjunto Comercial/Sala
3                    Apartamento
4                    Apartamento
                  ...           
32955                  Quitinete
32956                Apartamento
32957                Apartamento
32958                Apartamento
32959    Conjunto Comercial/Sala
Name: Tipo, Length: 32960, dtype: object

In [8]:
dataset['Tipo'].to_frame()

Tipo
0                    Quitinete
1                         Casa
2      Conjunto Comercial/Sala
3                  Apartamento
4                  Apartamento
...                        ...
32955                Quitinete
32956              Apartamento
32957              Apartamento
32958              Apartamento
32959  Conjunto Comercial/Sala

[32960 rows x 1 columns]

In [10]:
pd.DataFrame(dataset['Tipo'])

Tipo
0                    Quitinete
1                         Casa
2      Conjunto Comercial/Sala
3                  Apartamento
4                  Apartamento
...                        ...
32955                Quitinete
32956              Apartamento
32957              Apartamento
32958              Apartamento
32959  Conjunto Comercial/Sala

[32960 rows x 1 columns]

In [12]:
dataset[['Tipo']]

Tipo
0                    Quitinete
1                         Casa
2      Conjunto Comercial/Sala
3                  Apartamento
4                  Apartamento
...                        ...
32955                Quitinete
32956              Apartamento
32957              Apartamento
32958              Apartamento
32959  Conjunto Comercial/Sala

[32960 rows x 1 columns]

In [22]:
tipo_de_imoveis = dataset['Tipo'].to_frame()
tipo_de_imoveis.head()

Tipo
0                Quitinete
1                     Casa
2  Conjunto Comercial/Sala
3              Apartamento
4              Apartamento

In [23]:
tipo_de_imoveis['Tipo'].unique()

array(['Quitinete', 'Casa', 'Conjunto Comercial/Sala', 'Apartamento',
       'Casa de Condomínio', 'Prédio Inteiro', 'Flat', 'Loja/Salão',
       'Galpão/Depósito/Armazém', 'Casa Comercial', 'Casa de Vila',
       'Terreno Padrão', 'Box/Garagem', 'Loft',
       'Loja Shopping/ Ct Comercial', 'Chácara', 'Loteamento/Condomínio',
       'Sítio', 'Pousada/Chalé', 'Studio', 'Hotel', 'Indústria'],
      dtype=object)

In [24]:
tipo_de_imoveis = tipo_de_imoveis['Tipo'].drop_duplicates() # por padrão, seleciona o primeiro registro de cada tipo e remove os demais
tipo_de_imoveis

0                          Quitinete
1                               Casa
2            Conjunto Comercial/Sala
3                        Apartamento
7                 Casa de Condomínio
16                    Prédio Inteiro
17                              Flat
29                        Loja/Salão
80           Galpão/Depósito/Armazém
83                    Casa Comercial
117                     Casa de Vila
159                   Terreno Padrão
207                      Box/Garagem
347                             Loft
589      Loja Shopping/ Ct Comercial
2157                         Chácara
3354           Loteamento/Condomínio
4379                           Sítio
4721                   Pousada/Chalé
6983                          Studio
9687                           Hotel
23614                      Indústria
Name: Tipo, dtype: object

### Melhorando a apresentação dos Tipos de Imóveis

In [27]:
tipo_de_imoveis = tipo_de_imoveis.to_frame()
tipo_de_imoveis

Tipo
0                        Quitinete
1                             Casa
2          Conjunto Comercial/Sala
3                      Apartamento
7               Casa de Condomínio
16                  Prédio Inteiro
17                            Flat
29                      Loja/Salão
80         Galpão/Depósito/Armazém
83                  Casa Comercial
117                   Casa de Vila
159                 Terreno Padrão
207                    Box/Garagem
347                           Loft
589    Loja Shopping/ Ct Comercial
2157                       Chácara
3354         Loteamento/Condomínio
4379                         Sítio
4721                 Pousada/Chalé
6983                        Studio
9687                         Hotel
23614                    Indústria

In [28]:
tipo_de_imoveis.index

Int64Index([    0,     1,     2,     3,     7,    16,    17,    29,    80,
               83,   117,   159,   207,   347,   589,  2157,  3354,  4379,
             4721,  6983,  9687, 23614],
           dtype='int64')

In [30]:
tipo_de_imoveis.shape

(22, 1)

In [33]:
tipo_de_imoveis.index = range(tipo_de_imoveis.shape[0])

In [35]:
tipo_de_imoveis

Tipo
0                     Quitinete
1                          Casa
2       Conjunto Comercial/Sala
3                   Apartamento
4            Casa de Condomínio
5                Prédio Inteiro
6                          Flat
7                    Loja/Salão
8       Galpão/Depósito/Armazém
9                Casa Comercial
10                 Casa de Vila
11               Terreno Padrão
12                  Box/Garagem
13                         Loft
14  Loja Shopping/ Ct Comercial
15                      Chácara
16        Loteamento/Condomínio
17                        Sítio
18                Pousada/Chalé
19                       Studio
20                        Hotel
21                    Indústria

In [36]:
type(tipo_de_imoveis)

pandas.core.frame.DataFrame

### Filtrando apenas Imóveis Residenciais

In [39]:
list(tipo_de_imoveis['Tipo'])

['Quitinete',
 'Casa',
 'Conjunto Comercial/Sala',
 'Apartamento',
 'Casa de Condomínio',
 'Prédio Inteiro',
 'Flat',
 'Loja/Salão',
 'Galpão/Depósito/Armazém',
 'Casa Comercial',
 'Casa de Vila',
 'Terreno Padrão',
 'Box/Garagem',
 'Loft',
 'Loja Shopping/ Ct Comercial',
 'Chácara',
 'Loteamento/Condomínio',
 'Sítio',
 'Pousada/Chalé',
 'Studio',
 'Hotel',
 'Indústria']

In [41]:
tipos_residenciais = ['Quitinete', 'Casa', 'Apartamento', 'Casa de Condomínio', 'Casa de Vila']
tipos_residenciais

['Quitinete', 'Casa', 'Apartamento', 'Casa de Condomínio', 'Casa de Vila']

In [42]:
dataset.head()

Tipo           Bairro  Quartos  Vagas  Suites  Area  \
0                Quitinete       Copacabana        1      0       0    40   
1                     Casa  Jardim Botânico        2      0       1   100   
2  Conjunto Comercial/Sala  Barra da Tijuca        0      4       0   150   
3              Apartamento           Centro        1      0       0    15   
4              Apartamento     Higienópolis        1      0       0    48   

    Valor  Condominio    IPTU  
0  1700.0       500.0    60.0  
1  7000.0         NaN     NaN  
2  5200.0      4020.0  1111.0  
3   800.0       390.0    20.0  
4   800.0       230.0     NaN

In [44]:
# retorna uma Series de booleans informando se uma respectiva linha da tabela possui um Tipo
# que está dentro de (isin) --- é um dos elementos --- da lista/conjunto definidos em tipos_residenciais
dataset['Tipo'].isin(tipos_residenciais)

0         True
1         True
2        False
3         True
4         True
         ...  
32955     True
32956     True
32957     True
32958     True
32959    False
Name: Tipo, Length: 32960, dtype: bool

In [47]:
selecao = dataset['Tipo'].isin(tipos_residenciais)
dataset_residencial = dataset.loc[selecao]
dataset_residencial

Tipo                Bairro  Quartos  Vagas  Suites  Area  \
0        Quitinete            Copacabana        1      0       0    40   
1             Casa       Jardim Botânico        2      0       1   100   
3      Apartamento                Centro        1      0       0    15   
4      Apartamento          Higienópolis        1      0       0    48   
5      Apartamento          Vista Alegre        3      1       0    70   
...            ...                   ...      ...    ...     ...   ...   
32953  Apartamento                 Méier        2      0       0    70   
32955    Quitinete                Centro        0      0       0    27   
32956  Apartamento           Jacarepaguá        3      1       2    78   
32957  Apartamento  São Francisco Xavier        2      1       0    48   
32958  Apartamento                Leblon        2      0       0    70   

        Valor  Condominio  IPTU  
0      1700.0       500.0  60.0  
1      7000.0         NaN   NaN  
3       800.0       390.0  20.0  
4       800.0       230.0   NaN  
5      1200.0         NaN   NaN  
...       ...         ...   ...  
32953   900.0       490.0  48.0  
32955   800.0       350.0  25.0  
32956  1800.0       800.0  40.0  
32957  1400.0       509.0  37.0  
32958  3000.0       760.0   NaN  

[22580 rows x 9 columns]

Note que os índices, na tabela filtrada, não estão de 0 até o tamanho da tabela - 1. <br/>
Temos apenas 22580 linhas mas temos índices, p. ex, iguais a 32957, 32958.

In [48]:
dataset_residencial.index = range(dataset_residencial.shape[0])
dataset_residencial

Tipo                Bairro  Quartos  Vagas  Suites  Area  \
0        Quitinete            Copacabana        1      0       0    40   
1             Casa       Jardim Botânico        2      0       1   100   
2      Apartamento                Centro        1      0       0    15   
3      Apartamento          Higienópolis        1      0       0    48   
4      Apartamento          Vista Alegre        3      1       0    70   
...            ...                   ...      ...    ...     ...   ...   
22575  Apartamento                 Méier        2      0       0    70   
22576    Quitinete                Centro        0      0       0    27   
22577  Apartamento           Jacarepaguá        3      1       2    78   
22578  Apartamento  São Francisco Xavier        2      1       0    48   
22579  Apartamento                Leblon        2      0       0    70   

        Valor  Condominio  IPTU  
0      1700.0       500.0  60.0  
1      7000.0         NaN   NaN  
2       800.0       390.0  20.0  
3       800.0       230.0   NaN  
4      1200.0         NaN   NaN  
...       ...         ...   ...  
22575   900.0       490.0  48.0  
22576   800.0       350.0  25.0  
22577  1800.0       800.0  40.0  
22578  1400.0       509.0  37.0  
22579  3000.0       760.0   NaN  

[22580 rows x 9 columns]

### Exportando o Dataset Filtrado

In [49]:
# vamos salvar o dataset com apenas dados de imóveis residenciais
# vamos deixar o mesmo separador que tinha no dataset original
# exportando o dataset dessa maneira, o pandas criará uma coluna para representar os índices do DataFrame
dataset_residencial.to_csv('datasets/aluguel_residencial.csv', sep = ';')

In [50]:
# note a coluna Unnamed: 0
pd.read_csv('datasets/aluguel_residencial.csv', sep=';')

Unnamed: 0         Tipo                Bairro  Quartos  Vagas  Suites  \
0               0    Quitinete            Copacabana        1      0       0   
1               1         Casa       Jardim Botânico        2      0       1   
2               2  Apartamento                Centro        1      0       0   
3               3  Apartamento          Higienópolis        1      0       0   
4               4  Apartamento          Vista Alegre        3      1       0   
...           ...          ...                   ...      ...    ...     ...   
22575       22575  Apartamento                 Méier        2      0       0   
22576       22576    Quitinete                Centro        0      0       0   
22577       22577  Apartamento           Jacarepaguá        3      1       2   
22578       22578  Apartamento  São Francisco Xavier        2      1       0   
22579       22579  Apartamento                Leblon        2      0       0   

       Area   Valor  Condominio  IPTU  
0        40  1700.0       500.0  60.0  
1       100  7000.0         NaN   NaN  
2        15   800.0       390.0  20.0  
3        48   800.0       230.0   NaN  
4        70  1200.0         NaN   NaN  
...     ...     ...         ...   ...  
22575    70   900.0       490.0  48.0  
22576    27   800.0       350.0  25.0  
22577    78  1800.0       800.0  40.0  
22578    48  1400.0       509.0  37.0  
22579    70  3000.0       760.0   NaN  

[22580 rows x 10 columns]

In [51]:
# para corrigir isso, podemos dizer que não queremos salvar os índices
dataset_residencial.to_csv('datasets/aluguel_residencial.csv', sep = ';', index=False)

In [52]:
pd.read_csv('datasets/aluguel_residencial.csv', sep=';')

Tipo                Bairro  Quartos  Vagas  Suites  Area  \
0        Quitinete            Copacabana        1      0       0    40   
1             Casa       Jardim Botânico        2      0       1   100   
2      Apartamento                Centro        1      0       0    15   
3      Apartamento          Higienópolis        1      0       0    48   
4      Apartamento          Vista Alegre        3      1       0    70   
...            ...                   ...      ...    ...     ...   ...   
22575  Apartamento                 Méier        2      0       0    70   
22576    Quitinete                Centro        0      0       0    27   
22577  Apartamento           Jacarepaguá        3      1       2    78   
22578  Apartamento  São Francisco Xavier        2      1       0    48   
22579  Apartamento                Leblon        2      0       0    70   

        Valor  Condominio  IPTU  
0      1700.0       500.0  60.0  
1      7000.0         NaN   NaN  
2       800.0       390.0  20.0  
3       800.0       230.0   NaN  
4      1200.0         NaN   NaN  
...       ...         ...   ...  
22575   900.0       490.0  48.0  
22576   800.0       350.0  25.0  
22577  1800.0       800.0  40.0  
22578  1400.0       509.0  37.0  
22579  3000.0       760.0   NaN  

[22580 rows x 9 columns]

# Exercícios

1) Categorize os imóveis de acordo com sua área, seguindo os intervalos abaixo:
- 0–40: Pequeno
– 40–70: Médio
- 70–120: Grande
- 120–: Enrome

Mostre a proporção de cada intervalo de maneira tabular, e utilizando visualizações. Sugestões de gráficos: https://datavizcatalogue.com/

2) Refaça o exercício anterior, mas agora para cada tipo de imóvel. Compare as proporção de cada tipo.

3) Para cada bairro, mostre a média e desvio padrão do valor do aluguel, do número de quartos e da área do imóvel.
Selecione alguns bairros e os compare utilizando gráficos.